pkg/sessionctx/variable/session.go

```diff
@@ -1618,6 +1618,40 @@ type SessionVars struct {
 
 	// GroupConcatMaxLen represents the maximum length of the result of GROUP_CONCAT.
 	GroupConcatMaxLen uint64
+
+	// TablesBufferPool is a memory pool for table related memory allocation that aims to reuse memory
+	// and saves allocation
+	// The buffers are supposed to be used inside AddRecord/UpdateRecord/RemoveRecord.
+	// It's users duty to reset them before use.
+	TablesBufferPool TablesBufferPool
```

Comment:
How about just add a new method in `table.MutateContext` without modify `SessionVars`: https://github.com/pingcap/tidb/blob/1a24c032126dce4a79ea14b51108dbf4caf02a03/pkg/table/context/table.go#L32C6-L32C19

```
type MutateContext interface {
	// ...
	GetTablesBufferPool() * TablesBufferPool
	// ...
}

Reply:
You can implement this method here:

https://github.com/pingcap/tidb/blob/master/pkg/table/contextimpl/table.go